### PR TITLE
[TIMOB-19386] Rewrite project, add new API's, support bitcode, update travis

### PR DESCRIPTION
**Pull request by hansemannn**
JIRA: https://jira.appcelerator.org/browse/TIMOB-19386

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,4 +1,4 @@
-Copyright 2013 Appcelerator, Inc.
+Copyright 2015 Appcelerator, Inc.
```
Comment (cheekiatng) [inline LICENSE]: 2013-2015

File: apidoc/Accelerometer.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+---
+name: Modules.CoreMotion.Acceleratometer
```
Comment (cheekiatng) [inline apidoc/Accelerometer.yml]: spelling: Accelerometer

**Comment by cheekiatng** [comment]
Don't you need a
```
-(void)dealloc
{
    RELEASE_TO_NIL(accelerometer)
    RELEASE_TO_NIL(gyroscope)
    RELEASE_TO_NIL(magnetometer)
    RELEASE_TO_NIL(deviceMotion)
    RELEASE_TO_NIL(motionActivity)
    RELEASE_TO_NIL(stepCounter)
    [super dealloc];
}
```
in TiCoreMotionModule.m?

By the way, in TiCoreMotionModule.m and h, it'll be better if acceleratometer is spelt correctly as accelerometer

**Comment by cheekiatng** [comment]
https://github.com/hansemannn/ti.coremotion/blob/TIMOB-19386/iphone/Classes/CMHelper.m#L168
(NSArray<CMMotionActivity*>*)array may cause errors for Xcode 6.4

**Comment by cheekiatng** [comment]
https://github.com/hansemannn/ti.coremotion/blob/TIMOB-19386/iphone/Classes/TiCoremotionGyroscopeProxy.m#L25
This parameter is not optional. you should use ENSURE_TYPE or ENSURE_SINGLE_ARG instead of ENSURE_TYPE_OR_NIL

**Comment by cheekiatng** [comment]
same as above.
https://github.com/hansemannn/ti.coremotion/blob/TIMOB-19386/iphone/Classes/TiCoremotionMagnetometerProxy.m#L25

**Comment by cheekiatng** [comment]
you could ENSURE_TYPE(args, NSArray) here:
https://github.com/hansemannn/ti.coremotion/blob/TIMOB-19386/iphone/Classes/TiCoremotionStepCounterProxy.m#L30
and
https://github.com/hansemannn/ti.coremotion/blob/TIMOB-19386/iphone/Classes/TiCoremotionStepCounterProxy.m#L59

**Comment by cheekiatng** [comment]
TiCoremotionPedometerProxy.h and m physical files are in the root. It'll be nicer if they are in the Class folder

File: README.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # Core Motion [![Build Status](https://travis-ci.org/appcelerator-modules/ti.coremotion.svg?branch=master)](https://travis-ci.org/appcelerator-modules/ti.coremotion)
 
-**This module is available only for iOS 7 and above.**
+**This module is available only for iOS 7 and later.**
 
 ## COMPILING AND PACKAGING THE MODULE
```
Comment (cheekiatng) [inline README.md]: remove and change the entire description of the installation of the module here. the git submodule etc. was meant for hyper loop which is not applicable now that you have changed this module.

**Comment by hansemannn** [comment]
Addressed comments, thanks @cheekiatng !

**Comment by cheekiatng** [comment]
Functional Testing Based on Example Code:
getAccelerometerData and isAccelerometerActive not working.
Steps to reproduce:
1. startAccelerometerUpdates //no callback
2. check isAccelerometerActive. it returns false, should be true.
3. check getAccelerometerData. It has null data, should have data.

**Comment by cheekiatng** [comment]
Functional Testing Based on Example Code:
stopGyroUpdates crash.
Steps to reproduce:
1. startGyroUpdates(cb)
2. stopGyroUpdates or isGyroActive or getGyroData. this will crash.
Not sure if it's because the default sampling rate is too high? because if i do setMagnetometerUpdateInterval(1000) FIRST and repeat the above test, there's no crash.

**Comment by hansemannn** [comment]
> startAccelerometerUpdates //no callback

The method `startAcceleratometerUpdates` without callback don't fire any callback (native behavior). The user received the data from `getAccelerometerData`.

> check isAccelerometerActive. it returns false, should be true.

See above. It is only fired when the operation is on the queue = has an active. callback. Native behavior

> 3. check getAccelerometerData. It has null data, should have data.

Will check

**Comment by cheekiatng** [comment]
CR and FT Passed. APPROVED!

**Pull request merged**